pkg/externalauthproviders/flags.go

```diff
@@ -0,0 +1,351 @@
+		nil,
+		"A comma-separated list of audiences that the token was issued for.",
+	)
+
+	cmd.Flags().StringVar(
+		&args.issuerUrl,
+		"issuer-url",
+		"",
+		"The serving url of the token issuer.",
+	)
+
+	cmd.Flags().StringVar(
+		&args.issuerCaFile,
+		"issuer-ca-file",
+		"",
+		"Path to certificate file to use when making requests to the issuer server.",
+	)
+
+	cmd.Flags().StringVar(
+		&args.claimMappingGroupsClaim,
+		"claim-mapping-groups-claim",
+		"",
+		"Describes rules on how to transform information from an ID token into a cluster identity.",
+	)
+
+	cmd.Flags().StringVar(
+		&args.claimMappingUsernameClaim,
+		"claim-mapping-username-claim",
+		"",
+		"The name of the claim that should be used to construct usernmaes for the cluster identity.",
+	)
+
+	cmd.Flags().StringSliceVar(
+		&args.claimValidationRule,
+		"claim-validation-rule",
+		nil,
+		fmt.Sprintf("The input will be in a <claim>:<required_value> format. "+
+			"To have multiple claim validation rules, you could separate the values by ','. "+
+			"The input could be in a <claim>:<required_value>,<claim>:<required_value> format. "+
+			"Claim will be the rules that are applied to validate token claims to authenticate users. "+
+			"The required value is the token for the ClaimValidationRules."),
+	)
+
+	cmd.Flags().StringVar(
+		&args.consoleClientId,
+		"console-client-id",
+		"",
+		"The application or client id for your app registration that is used for console.",
+	)
+
+	cmd.Flags().StringVar(
+		&args.consoleClientSecret,
+		"console-client-secret",
+		"",
+		"The value of the client secret that is associated with your console app registration.",
+	)
+
+	return args
+}
+
+func GetExternalAuthOptions(
+	cmd *pflag.FlagSet, prefix string, confirmBeforeAllArgs bool, externalAuthProvidersArgs *ExternalAuthProvidersArgs,
+) (*ExternalAuthProvidersArgs, error) {
+
+	var err error
+	result := &ExternalAuthProvidersArgs{}
+
+	result.name = externalAuthProvidersArgs.name
+	result.issuerAudiences = externalAuthProvidersArgs.issuerAudiences
+	result.issuerUrl = externalAuthProvidersArgs.issuerUrl
+	result.issuerCaFile = externalAuthProvidersArgs.issuerCaFile
+	result.claimMappingGroupsClaim = externalAuthProvidersArgs.claimMappingGroupsClaim
+	result.claimMappingUsernameClaim = externalAuthProvidersArgs.claimMappingUsernameClaim
+	result.claimValidationRule = externalAuthProvidersArgs.claimValidationRule
+	result.consoleClientId = externalAuthProvidersArgs.consoleClientId
+	result.consoleClientSecret = externalAuthProvidersArgs.consoleClientSecret
+
+	issuerAudiencesSlice := externalAuthProvidersArgs.issuerAudiences
+	claimValidationRuleSlice := externalAuthProvidersArgs.claimValidationRule
+
+	if !interactive.Enabled() {
+		return nil, nil
+	}
+
+	if interactive.Enabled() && !cmd.Changed("name") {
+		result.name, err = interactive.GetString(interactive.Input{
+			Question: "Name",
+			Default:  result.name,
+			Help:     cmd.Lookup("name").Usage,
+			Required: true,
```

Comment:
I didn't add validations for many fields, due to not knowing if the validation are azure specific or it could accommodate other platforms. Will look more into the documentation.